torch/lib/THCUNN/generic/ClassNLLCriterion.cu

```diff
@@ -97,15 +125,16 @@ void THNN_(ClassNLLCriterion_updateGradInput)(
            THCState *state,
            THCTensor *input,
            THCIndexTensor *target,
+           THCTensor *gradOutput,
```

Comment:
If you're going to take in a `gradOutput` you need to make sure you use it in both the reduce and non-reduce cases.